src/odemis/driver/smaract.py

```diff
@@ -1189,8 +1189,8 @@ def __init__(self, name, role, locator, axes, ref_on_init=False, linear_speed=0.
         self._locator = locator
 
         # Be sure that the axes are defined in the controller
-        if not all(i in axes.keys() for i in ['x', 'y', 'z', 'rx', 'rz']):
-            raise ValueError("Not all axes are defined. Must have x, y, z, rx, rz")
+        if set(axes.keys()) < {'x', 'y', 'z', 'rx', 'rz'}:
+            raise ValueError("Some of the axes are not allowed, it should have only x, y, z, rx, rz")
```

Comment:
Sorry, I didn't read carefully the code after. If the user can just passes {"x"}, then the Component only provides 1 axis; "x". Is that correct? If so I think you'd better just force to always pass the 5 axes. This is a 5 DoF actuator! 
=> change the test to `set(axes.keys()) == {'x', 'y', 'z', 'rx', 'rz'}`.

If really it's handy to support modes where fewer axes are provided, then the test should be just inverted to:
`set(axes.keys()) > {'x', 'y', 'z', 'rx', 'rz'}:`. And maybe automatically put default values for the axes which are not specified?